mir_eval/chord.py

```diff
@@ -1342,6 +1347,136 @@ def sevenths_inv(reference_labels, estimated_labels):
     return comparison_scores
 
 
+def directional_hamming_distance(reference_intervals, estimated_intervals):
+    """Compute the directional hamming distance between reference and
+    estimated intervals as defined by [#harte2010towards]_ and used for MIREX
+    'OverSeg', 'UnderSeg' and 'MeanSeg' measures.
+
+    Examples
+    --------
+    >>> (ref_intervals,
+    ...  ref_labels) = mir_eval.io.load_labeled_intervals('ref.lab')
+    >>> (est_intervals,
+    ...  est_labels) = mir_eval.io.load_labeled_intervals('est.lab')
+    >>> overseg = 1 - mir_eval.chord.directional_hamming_distance(
+    ...     ref_intervals, est_intervals)
+    >>> underseg = 1 - mir_eval.chord.directional_hamming_distance(
+    ...     est_intervals, ref_intervals)
+    >>> meanseg = min(overseg, underseg)
+
+    Parameters
+    ----------
+    reference_intervals : np.ndarray, shape=(n, 2), dtype=float
+        Reference chord intervals to score against.
+    estimated_intervals : np.ndarray, shape=(m, 2), dtype=float
+        Estimated chord intervals to score against.
+
+    Returns
+    -------
+    directional hamming distance : float
+        directional hamming distance between reference intervals and
+        estimated intervals.
+    """
+    util.validate_intervals(estimated_intervals)
+    util.validate_intervals(reference_intervals)
+    est_ts = np.unique(estimated_intervals.flatten())
+    seg = 0.
+    for start, end in reference_intervals:
+        dur = end - start
+        between_start_end = est_ts[(est_ts >= start) & (est_ts < end)]
+        seg_ts = np.hstack([start, between_start_end, end])
+        seg += dur - np.diff(seg_ts).max()
+    return seg / (reference_intervals[-1, 1] - reference_intervals[0, 0])
```

Comment:
possible div-by-0 here, since `validate_intervals` doesn't check for non-emptiness.  Maybe add an explicit check here, and document the behavior if reference intervals are empty.

Reply:
I think it does implicitly in the following check (util.py:770):
```python
    if intervals.ndim != 2 or intervals.shape[1] != 2:
        raise ValueError('Intervals should be n-by-2 numpy ndarray, '
                         'but shape={}'.format(intervals.shape))
```
Here, `np.atleast_2d(np.array([])).shape[1] == 0`, and thus the `ValueError` will be raised. So, at least one interval must be present, and it must have a positive duration, and thus `reference_intervals[-1, 1] - reference_intervals[0, 0]` must be positive.

Reply:
Ah, quite correct, thanks.  Sorry I missed that before.

LGTM!